src/pyspiffe/config.py

```diff
@@ -0,0 +1,21 @@
+import os
+from typing import Dict
+
+SPIFFE_ENDPOINT_SOCKET = 'SPIFFE_ENDPOINT_SOCKET'
+
+
+class ConfigSetter:
+    def __init__(self):
+        self._apply_default_config()
+        self._apply_environment_variables()
+
+    def get_config(self) -> Dict:
+        return self._config
+
+    def _apply_default_config(self):
+        self._config = {
+            SPIFFE_ENDPOINT_SOCKET: None,
+        }
+
+    def _apply_environment_variables(self):
+        self._config[SPIFFE_ENDPOINT_SOCKET] = os.environ.get(SPIFFE_ENDPOINT_SOCKET)
```

Comment:
I think this should raise an additional exception with specific details, otherwise the exception isn't very specific on what's wrong and how to resolve - you'll get something like:

```
Traceback (most recent call last):
  File "/home/scott/.pyenv/versions/3.6.12/lib/python3.6/site-packages/pyspiffe/workloadapi/default_workload_api_client.py", line 34, in __init__
    self.spiffe_socket_path=os.environ['SPIFFE_ENDPOINT_SOCKET']
  File "/home/scott/.pyenv/versions/3.6.12/lib/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'SPIFFE_ENDPOINT_SOCKET'
```

Which isn't at all clear that this is for an environment variable, or that the variable is a requirement when called this way.

So in my stub implementation (where I implemented a temporary version of this until this code is in), I wrapped that in a RuntimeError (not sure if that's right, but that's what I did), so in addition to the above you get the following with details on what went wrong and how to fix:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 6, in <module>
    wlapi = DefaultWorkloadApiClient()
  File "/home/scott/.pyenv/versions/3.6.12/lib/python3.6/site-packages/pyspiffe/workloadapi/default_workload_api_client.py", line 36, in __init__
    raise RuntimeError('SPIFFE_ENDPOINT_SOCKET environment variable not specified to DefaultWorkloadApiClient')
```

Reply:
Hey @scotte, this line of code is using the `get` method from the dict type (`os.environ.get`), which doesn't raise an exception when the key doesn't exists, but returns a default value (`None` if not specified by the second parameter). In the other hand, when you try to access the dict by using the a key as an index it does raises an exception, as you stated in the example (`os.environ['SPIFFE_ENDPOINT_SOCKET']`).

Also, as we talked on the last py-spiffe sync meeting, the `SPIFFE_ENDPOINT_SOCKET` is not always mandatory, since a library user can programmatically define the socket by passing it as a parameter when instantiating the API client. So I don't think this is the right place to make such validation.

Reply:
I wasn't clear that I understand SPIFFE_ENDPOINT_SOCKET isn't a requirement when programmatically provided - but the point of this method is not to handle that full logic, but simply retrieve the variable if it exists.

So, I think you are right - while raising an exception if the socket isn't provided via any way, that logic probably belongs either in the WorkloadAPI implementation or a new method with the logic for this, but does not belong here. I would argue that as we currently only have one WorkloadAPI implementation anyway, the default one, we handle the runtime logic there and leave the library alone.

Additionally, it's possible that a different WorkloadAPI implementation would want to do something different anyway.

Long story short: It's good as is. 🏁